### DISCUSSION TITLE
JV-XP Editor Port Routing with Roland XP-30 and Ableton Live

### DISCUSSION THREAD

**Discussion by sinnersoid** [RomanKubiak/ctrlr#534] [Q&A]
NOTE: This issue occurs when using the JV-XP Editor _as a plugin in Ableton Live only_. The standalone version works fine for me.

An instance of JV-XP Editor v1.12 is dropped into a channel on Ableton Live 11.2.6. An RME Babyface is the MIDI interface and is multi-client. 

The MIDI settings in the editor are all the same as in its manual with the exception of using MIDI channel 15 instead of 1. The XP-30's MIDI channel for receiving and transmitting is also set to 15.

The MIDI preferences in Live are set as shown in Fig. 1 below. When selecting the Babyface for the input/output ports in the editor's MIDI menu, the error message in Fig. 2 is shown. Even though the green "Open OK" bar shows at the bottom, the error appears and the connection to the Roland XP-30 is not made.

![jvxp_error](https://user-images.githubusercontent.com/123333247/213942478-c9f3ac59-a7ff-4d82-94cc-e14c29ebd4ca.png)

If Live's MIDI ports to the Babyface (in Fig. 1) are all turned off, the connection to the XP-30 can be made ("ONLINE" appears in editor). However, now I cannot send MIDI data to Live via the XP-30's keyboard - or any other way.

From what I have read, this is because the MIDI port cannot be used for anything else but JV-XP Editor while the editor is in use.

If Live's MIDI ports to the Babyface (in Fig. 1) are all turned off, the connection to the XP-30 can be made ("ONLINE" appears in editor).

Because the interface is a Babyface, I only have one MIDI input port and one MIDI output port at my disposal. I certainly don't want to sacrifice all those MIDI channels' worth of possibilities for one plugin alone, even if it is a dream come true!

My question is this: can I use either a virtual MIDI port or MIDI port forwarding/routing software (or something similar) to have sysex and CC messages still go back and forth from XP-30 to the editor, thereby making the editor 'believe' it is occupying a different port than the Babyface's?

**Comment by damiensellier** [comment]
I never used the "midi from host" and midi comparator so I can't tell, but you should give it a try if your midi port/channel is already assigned to Live.

![host2midi](https://user-images.githubusercontent.com/55164385/214145932-8bf7874b-73fd-4e45-98ca-02639903dcbf.jpg)

It seems that ctrlr has issue sharing midi ports even with multi client interface.
This topic has been covered on the old ctrlr.org forum : 
https://ctrlr.org/forums/topic/working-standalone-but-not-in-cubase-as-plugin/

I had the same problem with a single-client midi interface and I had to create a virtual midi port with VirtualMidi or MidiOX:
https://www.tobias-erichsen.de/software/virtualmidi.html

And use a virtual midi patchbay and it works like a charm with Soundigy Midi Patchbay as a splitter : 
https://soundigy.com/midipatchbay.php
